src/main/java/com/gd/clinic/service/PatientService.java

```diff
@@ -0,0 +1,62 @@
+package com.gd.clinic.service;
+
+import com.gd.clinic.entity.Patient;
+import com.gd.clinic.repository.PatientRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+// TODO: Uncomment once security has been merged to this branch or vice versa
+//import org.springframework.security.core.context.SecurityContextHolder;
+import javax.transaction.Transactional;
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class PatientService {
+
+    private final PatientRepository patientRepository;
+
+    public Patient getById(UUID id) throws RuntimeException {
+        Optional<Patient> fetchedPatient = patientRepository.findById(id);
+        if(fetchedPatient.isPresent()) {
+            return fetchedPatient.get();
+        } else {
+            throw  new RuntimeException("Patient not found");
+        }
+    }
+
+    public boolean existsByInsuranceNumber(String in) {
+        return patientRepository.findOneByInsuranceNumber(in).isPresent();
```

Comment:
if you need this method only for this purpose, you can write repository method `existsByInsuranceNumber` that will return `boolean`

Reply:
It would, however its being used to fetch entities as well, so having a `boolean` there would only work when checking for presence but not to fetch data.